### ISSUE TITLE
Unbundle MongoDB from Deployd - move to it's own module

### ISSUE THREAD

**Issue by marshallswain** [deployd/deployd#317]
Reading on Jeffrey Cross's post here: https://plus.google.com/+JeffreyCross/posts/ijqF1XpRKEA

> Abstraction of Data Store so databases other than MongoDB can be used

As part of the Data Store abstraction, Deployd should no longer include MongoDB with the core, but should instead require setting up a database either in a server.js.  Since no other databases are currently supported, it could come "pre-wired" with a custom run script set up for a MongoDB instance running on localhost.

This would also eliminate issues like #98.

**Comment by dopry** [comment]
+1 for un-bundling mongo.
-1 for trying to make an odm/orm that works for both mongo and sql. I'd rather focus on better leveraging the features of mongodb, rather than being limited to the least common denominator of SQL and Mongo.

**Comment by marshallswain** [comment]
@dopry Good point on the least common denominator scenario.  We want to take full advantage of everything Mongo has to offer, nothing left out.

**Comment by ericfong** [comment]
Un-bundling mongo is very ideal. Seems to be too ideal to unbundling but still easy to code

**Comment by moorthi07** [comment]
How hard is it to integrate with jugglingdb?

**Comment by marshallswain** [comment]
@moorthi07, I don't think anybody has used any other database successfully and reported on it.  I've found FeathersJS to have probably the most extensive database support (I helped it get there).  It still doesn't have a GUI, but there are options in the works.
